packages/zwave-js/src/lib/commandclass/BarrierOperatorCC.ts

```diff
@@ -216,25 +211,66 @@ export class BarrierOperatorCCEventSet extends BarrierOperatorCC {
 				ZWaveErrorCodes.Deserialization_NotImplemented,
 			);
 		} else {
-			// TODO: Populate properties from options object
-			throw new Error("not implemented");
+			this.type = options.signalType;
+			this.state = options.signalState;
 		}
 	}
+	public type: SignalType;
+	public state: SignalState;
 
 	public serialize(): Buffer {
 		this.payload = Buffer.from([
-			/* TODO: serialize */
+			(this.payload[0] = this.type),
+			(this.payload[1] = this.state),
 		]);
 		return super.serialize();
 	}
 }
 
+@CCCommand(BarrierOperatorCommand.Report)
+export class BarrierOperatorCCEventReport extends BarrierOperatorCC {
+	public constructor(
+		driver: Driver,
+		options: CommandClassDeserializationOptions,
+	) {
+		super(driver, options);
+
+		validatePayload(this.payload.length >= 2);
+		this._type = this.payload[0];
+		this._state = this.payload[1];
+
+		this.persistValues();
+	}
+
+	private _type: SignalType | undefined;
```

Comment:
You're definitely setting a value in the constructor, so this should not include `| undefined`.